### PR TITLE
[release-1.23] add default retry policy for inbound routes

### PR DESCRIPTION
**Pull request by ramaraochavali**
Manual backport of https://github.com/istio/istio/pull/52055

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure
- [ ] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### CODE REVIEW

File: pilot/pkg/networking/core/route/route_test.go [hunk 1]
```diff
@@ -2963,3 +2966,70 @@ func TestSortVHostRoutes(t *testing.T) {
...
+				Decorator: &envoyroute.Decorator{
+					Operation: "operation",
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			test.SetForTest(t, &features.EnableInboundRetryPolicy, tc.enableRetry)
+			inroute := route.BuildDefaultHTTPInboundRoute(&model.Proxy{IstioVersion: &model.IstioVersion{Major: 1, Minor: 24, Patch: -1}},
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/route/route_test.go:3028]: is this right?

Comment (ramaraochavali) [reply pilot/pkg/networking/core/route/route_test.go]: it does not matter - but I fixed it

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**